### ISSUE TITLE
about training GREW using Baseline

### ISSUE THREAD

**Issue by King-Endeavor** [ShiqiYu/OpenGait#105]
## System information (version)

- Pytorch => :grey_question:
- Operating System / Platform => :grey_question:
- Cuda => :grey_question:

## Detailed description
GREW数据集按照文档处理，使用单卡3090训练
400Iteration后到现在10000Iteration损失一直为nan，triplet_loss_num围绕在403000左右，请问这正常吗？
[2023-01-30 12:35:51] [INFO]: Iteration 00100, Cost 116.61s, triplet_loss=0.2275, triplet_hard_loss=1.1326, triplet_loss_num=346738.4062, triplet_mean_dist=0.3253, softmax_loss=12.6231, softmax_accuracy=0.0003
[2023-01-30 12:37:21] [INFO]: Iteration 00200, Cost 90.22s, triplet_loss=0.2292, triplet_hard_loss=1.2973, triplet_loss_num=299717.0000, triplet_mean_dist=0.4165, softmax_loss=13.3734, softmax_accuracy=0.0003
[2023-01-30 12:38:54] [INFO]: Iteration 00300, Cost 92.76s, triplet_loss=0.4447, triplet_hard_loss=2.3350, triplet_loss_num=295953.2500, triplet_mean_dist=0.8897, softmax_loss=12.6169, softmax_accuracy=0.0006
[2023-01-30 12:42:49] [INFO]: Iteration 00400, Cost 234.99s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=397851.7188, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000
[2023-01-30 12:44:21] [INFO]: Iteration 00500, Cost 92.45s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=403821.4375, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0003
...........
[2023-01-30 16:00:15] [INFO]: Iteration 12900, Cost 95.40s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=404002.8750, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0003
[2023-01-30 16:01:50] [INFO]: Iteration 13000, Cost 94.83s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=403436.4062, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000
[2023-01-30 16:03:24] [INFO]: Iteration 13100, Cost 93.72s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=403703.0000, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000
[2023-01-30 16:04:56] [INFO]: Iteration 13200, Cost 91.80s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=403810.0312, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000
[2023-01-30 16:06:31] [INFO]: Iteration 13300, Cost 94.99s, triplet_loss=nan, triplet_hard_loss=nan, triplet_loss_num=402209.4375, triplet_mean_dist=nan, softmax_loss=nan, softmax_accuracy=0.0000

## Issue submission checklist
 - [ ] I checked the problem with documentation, FAQ, issues, and have not found solution.

**Comment by ChaoFan996** [comment]
Hi, we need to know the batch size you used. 
And more, a small batch size may lead to gradient oscillation cases that easily result in NaN.

**Comment by King-Endeavor** [comment]
batch_size是（32，8）除了修改数据集路径并没有修改Baseline_GREW.yaml 的内容

CUDA_VISIBLE_DEVICES=0 python -m torch.distributed.launch --nproc_per_node=1 opengait/main.py --cfgs ./configs/baseline/baseline_GREW.yaml --phase train --log_to_file

**Comment by King-Endeavor** [comment]
我把SGD换成Adam后，就可以正常训练，很奇怪。

**Comment by zhang123-sys** [comment]
GREW数据集可以通过百度网盘链接分享吗？

**Comment by noahshen98** [comment]
> GREW数据集可以通过百度网盘链接分享吗？

Hi, Zhang. You only access these datasets officially from the provider, and you can apply them in their regulation. We don't provide any data.

**Comment by King-Endeavor** [comment]
我又从头开始处理了一遍GREW数据集，现在使用SGD训练就没有出现上面的问题。
